pkg/run/gateway.go

```diff
@@ -305,6 +334,132 @@ func (s *BaseHttpGateway) handleApiHttpRequest(idx int) fasthttp.RequestHandler
 	}
 }
 
+// websocket request handler
+// TODO: Add broadcast capability
+func (s *BaseHttpGateway) handleWebsocketRequest(socketName string) func(ctx *fasthttp.RequestCtx) {
+	return func(ctx *fasthttp.RequestCtx) {
+		upgrader.CheckOrigin = func(ctx *fasthttp.RequestCtx) bool {
+			return true
+		}
+
+		connectionId := uuid.New().String()
+
+		connectionRequest := &v1.TriggerRequest{
+			Context: &v1.TriggerRequest_Websocket{
+				Websocket: &v1.WebsocketTriggerContext{
+					Socket:       socketName,
+					Event:        v1.WebsocketEvent_Connect,
+					ConnectionId: connectionId,
+				},
+			},
+		}
+
+		w, err := s.pool.GetWorker(&pool.GetWorkerOptions{
+			Trigger: connectionRequest,
+		})
+		if err != nil {
+			ctx.Error("No worker found to handle connection request", 404)
+			return
+		}
+
+		res, err := w.HandleTrigger(context.TODO(), connectionRequest)
+		// handshake error...
+		if err != nil {
+			return
+		}
```

Comment:
Should this error case be logged or returned?

Reply:
logged, will pop that in

Reply:
done